### PR TITLE
[dev-launcher] Fix Dev Client crash when server URL is missing the protocol

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Closes ENG-7517

When calling `replaceEXPScheme` with an `url` that does not include a scheme, `URLComponents.init()` returns `nil` which was crashing the app because we were force-unwrapping the result with the `!` operator.

# How

This PR updates the `replaceEXPScheme` function to no longer force-unwrap and return the original `url` when `URLComponents.init()` returns `nil`, which is the approach already used for Android:
https://github.com/expo/expo/blob/98d90433bde67adbff25cdbfdc9e774ce1b4d205/packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/helpers/DevLauncherURLHelper.kt#L5

One thing that we could do to improve the DX is to default the scheme of URLs provided to `loadApp` to `http`

# Test Plan

Tests using `bare-expo` with ` BOOL useDevClient = YES;` 
  

https://user-images.githubusercontent.com/11707729/219722351-7dabbae5-9ab3-45b5-a7a8-f9a64b911dfe.mov

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift [hunk 1]
```diff
@@ -65,11 +65,13 @@ public class EXDevLauncherURLHelper: NSObject {
...
   @objc
   public static func replaceEXPScheme(_ url: URL, to scheme: String) -> URL {
-    var components = URLComponents.init(url: url, resolvingAgainstBaseURL: false)!
-    if components.scheme == "exp" {
-      components.scheme = scheme
+    if var components = URLComponents(url: url, resolvingAgainstBaseURL: false) {
+      if components.scheme == "exp" {
+        components.scheme = scheme
+      }
+      return components.url!
```
Comment (douglowder) [inline packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift:72]:
```suggestion
      return components.url ?? ""
```

**Review by douglowder — CHANGES_REQUESTED** [review]
Made a suggestion to fix the SwiftLint violation

File: packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift (hunk 1 above)
Comment (gabrieldonadel) [reply packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift]: Updated the whole function so we no longer force unwrap and just default to the input url

Comment (douglowder) [reply packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift]: Actually this is a URL not a String, sorry I read the code too quickly. Let me have another look

**Review by douglowder — APPROVED** [review]

File: packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift (hunk 1 above)
Comment (douglowder) [reply packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift]: Your changes look good now, approved

**Pull request merged**